components/movementsensor/gpsrtkserial/gpsrtkserial_test.go

```diff
@@ -85,3 +89,34 @@ func TestReconfigure(t *testing.T) {
 	test.That(t, g.writePath, test.ShouldResemble, "/dev/ttyUSB1")
 	test.That(t, g.wbaud, test.ShouldEqual, 115200)
 }
+
+func TestPosition(t *testing.T) {
+
+	t.Run("position test with last error and no last position", func(t *testing.T) {
+		g := &rtkSerial{
+			err: movementsensor.NewLastError(1, 1),
+		}
+		g.err.Set(errors.New("last error test"))
+
+		point, flt, err := g.Position(context.Background(), nil)
+		test.That(t, movementsensor.IsPositionNaN(point), test.ShouldBeTrue)
+		test.That(t, math.IsNaN(flt), test.ShouldBeTrue)
+		test.That(t, err, test.ShouldBeError, "last error test")
+	})
+
+	t.Run("position test with last error and last position", func(t *testing.T) {
+		g := &rtkSerial{
+			err:          movementsensor.NewLastError(1, 1),
+			lastposition: movementsensor.NewLastPosition(),
+		}
+		g.lastposition.SetLastPosition(geo.NewPoint(42.1, 123))
```

Comment:
Can you add another test with invalid last position?

Reply:
yup, will do